ankaios_sdk/_components/complete_state.py

```diff
@@ -73,22 +69,45 @@
 from .workload import Workload
 from .workload_state import WorkloadStateCollection
 from .manifest import Manifest
-from ..utils import SUPPORTED_API_VERSION
+from ..utils import SUPPORTED_API_VERSION, _to_config_item
 
 
 class CompleteState:
     """
     A class to represent the complete state.
     """
-    def __init__(self) -> None:
+    def __init__(self,
+                 manifest: Manifest = None,
+                 configs: dict = None,
+                 workloads: list[Workload] = None,
+                 _proto: _ank_base.CompleteState = None
+                 ) -> None:
         """
-        Initializes an empty CompleteState instance with the given API version.
+        Initializes a CompleteState instance with the provided data.
+
+        Args:
+            manifest (Manifest): The manifest to initialize the complete state.
+            configs (dict): The configurations to set in the complete state.
+            workloads (list[Workload]): The workloads to set
+                in the complete state.
+            _proto (_ank_base.CompleteState): The proto message to initialize
+                the complete state.
         """
         self._complete_state = _ank_base.CompleteState()
         self._set_api_version(SUPPORTED_API_VERSION)
-        self._workloads: list[Workload] = []
-        self._workload_state_collection = WorkloadStateCollection()
-        self._configs = {}
+        if manifest:
+            self._complete_state.desiredState.CopyFrom(
+                manifest._to_desired_state()
+            )
+        if configs:
+            self.set_configs(configs)
+        if workloads:
+            self._complete_state.desiredState.workloads.workloads.clear()
+            for workload in workloads:
+                self._complete_state.desiredState.workloads.workloads[
+                    workload.name].CopyFrom(workload._to_proto())
+        if _proto:
+            self._complete_state = _proto
```

Comment:
In order to keep consistency between SDKs, wouldn't it help to enable just one method of initialization, as in Rust SDK?
```suggestion
        if manifest:
            self._complete_state.desiredState.CopyFrom(
                manifest._to_desired_state()
            )
            pass
        if configs:
            self.set_configs(configs)
            pass
        if workloads:
            self._complete_state.desiredState.workloads.workloads.clear()
            for workload in workloads:
                self._complete_state.desiredState.workloads.workloads[
                    workload.name].CopyFrom(workload._to_proto())
                    pass
        if _proto:
            self._complete_state = _proto
```
or maybe fix this by using `Union` from `typing`.

Reply:
In the Rust SDK, because of the design of the language, I need multiple constructors for each type, and the From trait can be combined with the constructors, which makes it very easy to use. In python, the pythonic way would be to have a single constructor with multiple types.

If I am to have a Union with all the types, I would need to improve the documentation to show exactly what can be there, and also to implement checks for each type (because python doesn't do it by itself). The current way is the easiest. 

I can return in each if, to make it more like a C++ switch, so that if you provide multiple variables, it does not execute multiple branches, but there might be the case where you provide workloads and configs as well, and this will not work anymore with the returns. It indeed falls on the user to use it as it's supposed to, but if he does not, there is nothing bad that can happen.

Reply:
As discussed, using `return` will provide the best solution to this.